### PR TITLE
Add Series float predicates: is_finite/1, is_infinite/1, is_nan/1

### LINKED ISSUE #426: Add Series float predicates:  is_finite/1, is_infinite/1, is_nan/1
They are float only.

### PR DESCRIPTION
**Pull request by sasikumar87**
This PR closes #426

### CODE REVIEW

File: lib/explorer/series.ex [hunk 1]
```diff
@@ -2732,6 +2732,72 @@ defmodule Explorer.Series do
...
+      iex> s3 = Explorer.Series.divide(s1, s2)
+      iex> Explorer.Series.is_finite(s3)
+      #Explorer.Series<
+        Polars[4]
+        boolean [false, true, false, nil]
+      >
+  """
+  @doc type: :element_wise
+  @spec is_finite(Series.t()) :: Series.t()
+  def is_finite(%Series{dtype: dtype} = series) when numeric_dtype?(dtype),
```
Comment (josevalim) [inline lib/explorer/series.ex:2753]: Do they actually work with integers? Or floats only? If they work with integers, can you add a test to `series_test.exs` for at least one of them? Thank you! :heart:

Comment (sasikumar87) [reply lib/explorer/series.ex:2753]: ah, I overlooked. It works only with float. Let me update.

File: lib/explorer/series.ex [hunk 2]
```diff
@@ -2732,6 +2732,72 @@ defmodule Explorer.Series do
...
+      iex> s3 = Explorer.Series.divide(s1, s2)
+      iex> Explorer.Series.is_finite(s3)
+      #Explorer.Series<
+        Polars[4]
+        boolean [false, true, false, nil]
+      >
+  """
+  @doc type: :element_wise
+  @spec is_finite(Series.t()) :: Series.t()
+  def is_finite(%Series{dtype: dtype} = series) when dtype == :float,
```
Comment (josevalim) [inline lib/explorer/series.ex:2753]:
```suggestion
  def is_finite(%Series{dtype: :float} = series),
```

File: lib/explorer/series.ex [hunk 3]
```diff
@@ -2732,6 +2732,72 @@ defmodule Explorer.Series do
...
+      iex> s3 = Explorer.Series.divide(s1, s2)
+      iex> Explorer.Series.is_infinite(s3)
+      #Explorer.Series<
+        Polars[4]
+        boolean [true, false, false, nil]
+      >
+  """
+  @doc type: :element_wise
+  @spec is_infinite(Series.t()) :: Series.t()
+  def is_infinite(%Series{dtype: dtype} = series) when dtype == :float,
```
Comment (josevalim) [inline lib/explorer/series.ex:2774]:
```suggestion
  def is_infinite(%Series{dtype: :float} = series),
```

File: lib/explorer/series.ex [hunk 4]
```diff
@@ -2732,6 +2732,72 @@ defmodule Explorer.Series do
...
+      iex> s3 = Explorer.Series.divide(s1, s2)
+      iex> Explorer.Series.is_nan(s3)
+      #Explorer.Series<
+        Polars[4]
+        boolean [false, false, true, nil]
+      >
+  """
+  @doc type: :element_wise
+  @spec is_nan(Series.t()) :: Series.t()
+  def is_nan(%Series{dtype: dtype} = series) when dtype == :float,
```
Comment (josevalim) [inline lib/explorer/series.ex:2796]:
```suggestion
  def is_nan(%Series{dtype: :float} = series),
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**